### DISCUSSION TITLE
What tutorial format for C++, Julia and Matlab?

### DISCUSSION THREAD

**Discussion by rubiop** [MeasureTransport/MParT-examples#26] [General]
For Python we use jupytext to produce notebooks.

What's the plan for the other languages?

I think that even if we won't have examples in the other languages in the documentation, I think it will be beneficial to have them in a notebook format in the git repo.

For matlab I was thinking about using the live code file format (.mlx) for the sake of simplicity. We will be able to run examples on matlab online.

What's the plan for Julia and C++? Is there an interest/easy way to run notebooks with C++?

**Comment by dannys4** [comment]
For c++- There's cling which I've used for notebooks in the past with moderate success. I have not plotted with c++ before, so I'm not really sure how that would work, but it's not too tricky to set up the notebook itself if we want to just show the results by printing, e.g., the sample KL divergence before and after optimization.

For Julia, I probably would just do Pluto, if only because it just stores everything in a .jl file, making version control easy enough.

**Reply by bonh** [reply]
+1 for Pluto

**Comment by bonh** [comment]
I don't think we should bother with C++ examples as notebooks.
